controller/src/beerocks/cli/beerocks_cli_bml.cpp

```diff
@@ -1806,5 +1913,63 @@ int cli_bml::get_dcs_scan_results(const std::string &radio_mac, unsigned int max
 {
     std::cout << "get_dcs_scan_results" << std::endl;
 
+    if (max_results_size == 0) {
+        std::cout << __func__ << "invalid input, max_results_size==0" << std::endl;
+        return -1;
+    }
+
+    struct BML_DCS_NEIGHBOR_AP *results = nullptr;
+
+    results =
+        (struct BML_DCS_NEIGHBOR_AP *)calloc(max_results_size, sizeof(struct BML_DCS_NEIGHBOR_AP));
```

Comment:
Why didn't you declare it on the stack???
Anyway, this is C allocation. The CPP way is with `new`/`delete` and the better option will be to use smart pointers.

Reply:
@tomereli already mentioned it - I fixed it and am using a user-defined-sized array declared on the stack.